### PR TITLE
[Refactor] Temp Quick All Refactor & Test

### PR DESCRIPTION
**Pull request by ParaskP7**
Tests: [AINFRA-51](https://linear.app/a8c/issue/AINFRA-51)

---

`TODO`

### CODE REVIEW

File: encryptedlogging/build.gradle.kts [hunk 1]
```diff
@@ -56,16 +61,20 @@ fun loadPropertiesFromFile(file: File): Properties {
 dependencies {
     implementation(libs.androidx.core.ktx)
     implementation(libs.androidx.appcompat)
+    implementation(libs.androidx.room.runtime)
+    ksp(libs.androidx.room.compiler)
+    api(libs.androidx.room.ktx)
```
Comment (wzieba) [inline encryptedlogging/build.gradle.kts:66]: 👋 What's the reason for making this `api`? I don't think that "encrypted logging" library should need its client to do anything regarding the databasae - if current design enforces `api` here, let's refactor the solution.

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/persistence/EncryptedLogDatabase.kt [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+)
+internal abstract class EncryptedLogDatabase : RoomDatabase() {
+    internal abstract val encryptedLogDao: EncryptedLogDao
+
+    companion object {
+        fun buildDb(applicationContext: Context) = Room.databaseBuilder(
+            applicationContext,
+            EncryptedLogDatabase::class.java,
+            "encrypted-log.db"
+        ).allowMainThreadQueries()
```
Comment (wzieba) [inline encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/persistence/EncryptedLogDatabase.kt:31]: For this small library, I think we can assert no main thread calls

**Review by wzieba — COMMENTED** [review]
Actually, WDYT about using a text file instead of a database for this? Text file is simpler, less dependencies and no migration overhead.

File: encryptedlogging/schemas/com.automattic.encryptedlogging.persistence.EncryptedLogDatabase/1.json [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+          {
+            "fieldPath": "filePath",
+            "columnName": "filePath",
+            "affinity": "TEXT"
+          },
+          {
+            "fieldPath": "dateCreated",
+            "columnName": "dateCreated",
+            "affinity": "TEXT"
+          },
```
Comment (wzieba) [inline encryptedlogging/schemas/com.automattic.encryptedlogging.persistence.EncryptedLogDatabase/1.json:26]:
```suggestion
  
```

to fix #12 :)

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/store/EncryptedLogStore.kt [hunk 4]
```diff
@@ -53,41 +42,21 @@ private const val MAX_RETRY_COUNT = 3
 private const val HTTP_STATUS_CODE_500 = 500
 private const val HTTP_STATUS_CODE_599 = 599
 
-internal class EncryptedLogStore constructor(
+internal class EncryptedLogStore(
```
Comment (wzieba) [inline encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/store/EncryptedLogStore.kt:40]: We'll have to assert, library-level, that this class will be a singleton as it holds a state 👍 .

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt [hunk 5]
```diff
@@ -52,18 +44,11 @@ public class AutomatticEncryptedLogging(
...
     }
 
-    @Suppress("unused")
-    @Subscribe(threadMode = ThreadMode.ASYNC)
-    internal fun onEncryptedLogUploaded(event: OnEncryptedLogUploaded) {
-        uploadState.value = event
-    }
-
-    override fun enqueueSendingEncryptedLogs(
+    override suspend fun enqueueSendingEncryptedLogs(
```
Comment (wzieba) [inline encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt:48]: We can do this or, to keep API contract intact, have our own Coroutine scope and allow clients to have an easier, non-coroutines API

**Comment by ParaskP7** [comment]
> Actually, WDYT about using a text file instead of a database for this? Text file is simpler, less dependencies and no migration overhead.

Yea @wzieba , I was thinking about it too, why not us using the [Preferences DataStore](https://developer.android.com/jetpack/androidx/releases/datastore) instead (if that's what you mean by test file). I wasn't sure why exactly we want (or wanted) multiple database entries, although we always limit the database result to 1 (#14), and did that even before that change via `firstOrNull()`. Plus, using `DATE_CREATED` to order the result so that the first log that's queued to get the priority for upload. This got me a bit confused, but I am up for a complete rewrite of this whole logic, if that's indeed a good moment for us to do so. 🤔

FYI: To be honest, while I was testing the library on DOAndroid, I did see 2 entries appear in the database, like when putting the device offline, etc. 🤷

File: encryptedlogging/build.gradle.kts (hunk 1 above)
Comment (ParaskP7) [reply encryptedlogging/build.gradle.kts:66]: No reason @wzieba, this is just a quick copy-paste (mistake) I did while getting this dependencies as is from another project (I think WCAndroid), just to test this asap, noted and thanks! 🔖

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/persistence/EncryptedLogDatabase.kt (hunk 2 above)
Comment (ParaskP7) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/persistence/EncryptedLogDatabase.kt:31]: Oh definitely, I agree @wzieba, another copy-paste (mistake) while me getting this config from another project (I think WCAndroid), noted and thanks! 🔖

File: encryptedlogging/schemas/com.automattic.encryptedlogging.persistence.EncryptedLogDatabase/1.json (hunk 3 above)
Comment (ParaskP7) [reply encryptedlogging/schemas/com.automattic.encryptedlogging.persistence.EncryptedLogDatabase/1.json:26]: Yea, I remember us discussing it @wzieba, I just kept it there because this `DATE_CREATED` is used on the database query to get the encrypted log for upload and I wasn't too sure about it. I was indeed seeing 2 such database entries once in a while during testing this library on DOAndroid, especially while putting the device offline. 🤔

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/store/EncryptedLogStore.kt (hunk 4 above)
Comment (ParaskP7) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/store/EncryptedLogStore.kt:40]: Aw nice, this is indeed a good catch @wzieba, I agree, let's make it a singleton! 🔖

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt (hunk 5 above)
Comment (ParaskP7) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt:48]: Yea, I was thinking about it too @wzieba, like moving this `CoroutineScope(Dispatchers.IO).launch { ... }` here (from where it was before). I was also thinking whether we want to do that for all current public `suspend` APIs, like `resetUploadStates()` and `uploadEncryptedLogs()`, wdyt? 🤔

Comment (wzieba) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt:48]: > I was also thinking whether we want to do that for all current public suspend APIs

I think that's a good idea, let's make this SDK as easy as possible to use for clients 👍

**Comment by wzieba** [comment]
I didn't mean `DataStore` but a plain text file.

>  I wasn't sure why exactly we want (or wanted) multiple database entries

Because the device can log things while it's offline, so we want to keep multiple encrypted logs to be quaried later.

> although we always limit the database result to 1 (https://github.com/Automattic/EncryptedLogging/pull/14)

It's only limited for sending part. We send one by one, but it doesn't mean we don't send all queued up logs. See: https://github.com/Automattic/EncryptedLogging/pull/14/files#diff-f19558ec3834c828a203033981838e9549bf699c0c5f1956c9494ae256ab3aaaR142

**Comment by ParaskP7** [comment]
> I didn't mean DataStore but a plain text file.

Got it, just wanted to make sure of it. I remember you doing something similar on a project (maybe on [parsely-android](https://github.com/Parsely/parsely-android)), not sure if I remember correctly... 🤔

> Because the device can log things while it's offline, so we want to keep multiple encrypted logs to be quaried later.

👍

> It's only limited for sending part. We send one by one, but it doesn't mean we don't send all queued up logs. See: https://github.com/Automattic/EncryptedLogging/pull/14/files#diff-f19558ec3834c828a203033981838e9549bf699c0c5f1956c9494ae256ab3aaaR142

Ah, okay, this is interesting, I somehow missed this. 💭

Then, just because of that, tbh, I wouldn't change that to plain text file and keep it as is, at least for this iteration. The purpose we are doing this refactor is to get rid of EventBus and WellSQL, maybe we shouldn't do an overhaul change and totally update the logic, from database to plain text file. 🤔

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt (hunk 5 above)
Comment (ParaskP7) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt:48]: Agreed! 🤝

FYI: Client apps would still need to use their own `CoroutineScope` (like `applicationScope`) for the flow API. me thinking of `observeEncryptedLogsUploadResult()`, but (I guess) this is another story altogether.

**Comment by wzieba** [comment]
> maybe we shouldn't do an overhaul change and totally update the logic, from database to plain text file. 🤔

I think a migration to a file is the same as to Room: just another implementation of persistence layer. The logic wouldn't change. I find it a little overcomplex to have a database that holds at most a few simple pointers, but if you prefer, we can think about it in another iteration.

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt (hunk 5 above)
Comment (wzieba) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt:48]: Actually, a thought to consider: `observeEncryptedLogsUploadResult` is not actually needed for clients. It's something this library inherited from in-app implementation.

This method is only used for logging purposes

https://github.com/bloom/DayOne-Android/blob/79c16e3025acceb26117ab509739a336c4799f18/app/src/main/java/com/dayoneapp/dayone/utils/EncryptedLoggingInitializer.kt#L22-L37

let's simply log success/failure inside the sdk.

Comment (ParaskP7) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt:48]: That's a good idea, I'll take a look and consider it next week, my returning from support rotation, thanks @wzieba ! 🔖

**Comment by ParaskP7** [comment]
> I think a migration to a file is the same as to Room: just another implementation of persistence layer. The logic wouldn't change.

👍

> I find it a little overcomplex to have a database that holds at most a few simple pointers...

I do understand where you're coming from @wzieba . 💯

> ...but if you prefer, we can think about it in another iteration.

Exactly this, I think we better do that on another iteration and keep the changes here to minimum, just to progress faster with this [AINFRA-539](https://linear.app/a8c/issue/AINFRA-539/) + [AINFRA-505](https://linear.app/a8c/issue/AINFRA-505) main issues of ours. 😊

FYI: I do think us creating another implementation of persistence layer would mean more work/testing for us (maybe even bug fixing), while Room provides a lot with zero effort/risk. The only downside of using Room atm is that it brings with it a set of extra dependencies for all our client. But, having said that, most if not all client are already using Room, one way or another, so I think this is not big of a deal... 🤷

**Comment by ParaskP7** [comment]
FYI: Closing this since it got opened only for testing/review purposes. Cc @wzieba

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt (hunk 5 above)
Comment (ParaskP7) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt:48]: @wzieba about `observeEncryptedLogsUploadResult()` and the fact that it is not actually needed for clients, I did some more research and found out that:
- `PCAndroid`: Not uses it at all.
- `DOAndroid`: Only uses it for [logging purposes](https://github.com/bloom/DayOne-Android/blob/trunk/app/src/main/java/com/dayoneapp/dayone/utils/EncryptedLoggingInitializer.kt#L22-L37).
- `WCAndroid`: Uses it for [logging and tracking purposes](https://github.com/woocommerce/woocommerce-android/blob/3242df4b93fbccb4fc39922d2d80b3d91b27ceeb/WooCommerce/src/main/kotlin/com/woocommerce/android/util/encryptedlogging/ObserveEncryptedLogsUploadResult.kt#L19).
- `JP/WPAndroid`: ~~Not uses it at all.~~ Uses it for [logging and tracking purposes](https://github.com/wordpress-mobile/WordPress-Android/blob/trunk/WordPress/src/main/java/org/wordpress/android/util/EncryptedLogging.kt#L72-L91).

As such, and just because of `WCAndroid`, I still think that it is better to still give this opportunity to clients, for them to do their own logging and tracking, instead of us doing that for them, should they decide to. Wdty? 🤔

---

FYI: As part of #18 and [this commit](https://github.com/Automattic/EncryptedLogging/pull/18/commits/d434b7db068e44e70cc4805774dc71e7d4adaeae) I actually removed the `AppLog` functionality from this library, replacing it with plain Android `Log`, and still think it is the right move forward.

PS: Similarly, I wouldn't be comfortable if we add `AnalyticsTracker` and bring [another library](https://github.com/Automattic/Automattic-Tracks-Android) into this project, that is, should we decide to do the tracking for them too.

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/store/EncryptedLogStore.kt (hunk 4 above)
Comment (ParaskP7) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/store/EncryptedLogStore.kt:40]: FYI: This was addresses in #19 and this commit: https://github.com/Automattic/EncryptedLogging/pull/19/commits/5efb1246bd42b42bd4f88fd5da4e72de2700e67a

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt (hunk 5 above)
Comment (ParaskP7) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt:48]: FYI: This was addresses in #19 and this commit: https://github.com/Automattic/EncryptedLogging/pull/19/commits/5bc1a2cc42f9d823d94f67bda504fe50839dda2c

Comment (wzieba) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt:48]: re: `observeEncryptedLogsUploadResult()`

I strongly believe that `WCAndroid` doesn't use data from tracking if encrypted logs were sent. I think we can remove this, and `observeEncryptedLogsUploadResult`, and simply log success/failure with logcat inside the library.

Comment (ParaskP7) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt:48]: FYI: I just asked WCAndroid team here: p1750327131116929-slack-CGPNUU63E

File: encryptedlogging/schemas/com.automattic.encryptedlogging.persistence.EncryptedLogDatabase/1.json (hunk 3 above)
Comment (ParaskP7) [reply encryptedlogging/schemas/com.automattic.encryptedlogging.persistence.EncryptedLogDatabase/1.json:26]: FYI: This was addresses in #20 and this commit: https://github.com/Automattic/EncryptedLogging/pull/20/commits/ef2bdcf0cca59bf0cc0b15d48866666d6f48b0fa

File: encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt (hunk 5 above)
Comment (ParaskP7) [reply encryptedlogging/src/main/kotlin/com/automattic/encryptedlogging/AutomatticEncryptedLogging.kt:48]: Related: [20#issuecomment](https://github.com/Automattic/EncryptedLogging/pull/20#issuecomment-2999808873)
